### PR TITLE
build: use PythonCall instead of PyCall

### PR DESCRIPTION
**Pull request by thazhemadam**
Migrate to using `PythonCall` instead of `PyCall`.

### CODE REVIEW

File: src/graph_building.jl [hunk 1]
```diff
@@ -118,21 +118,22 @@ end
...
-            for image_ind = 1:size(conn)[3]
+            for image_ind in range(0, stop = length(conn[0][0]) - 1)
                 # only add as neighbor if atom is not current center one AND there is connectivity to image
-                if (atom_ind != image_ind) && (conn[atom_ind, nb_ind, image_ind] != 0)
-                    weight_mat[atom_ind, nb_ind] +=
-                        conn[atom_ind, nb_ind, image_ind] / maximum(conn[atom_ind, :, :])
+                if (atom_ind != image_ind) && (pyconvert(Float64, (conn[atom_ind][nb_ind][image_ind])) != 0)
+                    conn_matrix = pyconvert(Matrix, conn[atom_ind])
+                    weight_mat[atom_ind + 1, nb_ind + 1] +=
+                        pyconvert(Float32, conn[atom_ind][nb_ind][image_ind]) / maximum(conn_matrix)
```
Comment (rkurchin) [inline src/graph_building.jl:142]: why do we need to do the `pyconvert` again here? Can we not just index into `conn_matrix` instead?

File: Project.toml [hunk 2]
```diff
@@ -23,13 +23,11 @@ Xtals = "ede5f01d-793e-4c47-9885-c447d1f18d6d"
 Cairo = "1"
 ChemistryFeaturization = "0.7"
 Colors = "0.12"
-Conda = "1.6"
```
Comment (rkurchin) [inline Project.toml:26]: we should probably add compats for CondaPkg and PythonCall, right?

**Review by rkurchin — COMMENTED** [review]
Generally LGTM. I made a couple small comments/suggestions. I'll also add the AtomsIO stuff on this branch and we can merge that in together, since that was part of the purpose of this change. Should hopefully have time to do that later today.

File: Project.toml (hunk 2 above)
Comment (rkurchin) [reply Project.toml:26]: I just did it myself

**Comment by rkurchin** [comment]
Made the other changes I wanted to make. I debated if I should just take out the Xtals.jl dependency, but decided to keep it, rationalizing that this is already sort of a "glue" package anyway so it doesn't seem worth splitting it to its own package just to remove that one dep, and reducing its functionality and/or putting it on those devs to support the functions we need seemed excessive. So anyway, it's there for now, but will only be used if the user explicitly passes in a `Crystal` object, and otherwise we're AtomsBase all the way down! 🎉 

The tests are giving a bunch of these warnings now:
```julia
┌ Warning: `ustrip(A::AbstractArray{T}) where T <: Number` is deprecated, use `ustrip.(A)` instead.
│   caller = macro expansion at broadcast.jl:140 [inlined]
└ @ Core ~/.julia/packages/StaticArrays/jA1zK/src/broadcast.jl:140
┌ Warning: `ustrip(A::AbstractArray{T}) where T <: Number` is deprecated, use `ustrip.(A)` instead.
│   caller = _broadcast_getindex_evalf at broadcast.jl:670 [inlined]
└ @ Core ./broadcast.jl:670
```
which I assume we can ignore; I'm guessing they're coming from inside of some AtomsIO thing that we probably don't need to worry about.

Anyway, @thazhemadam LMK if you see any issues with this, otherwise we can merge. I'd like to have our little showcase depend on this version so that the AtomsBase stuff is there...

**Review by rkurchin — APPROVED** [review]
LGTM, she says to herself

**Comment by rkurchin** [comment]
(I jankily merged Dhairya's old PR and then rebased this so that I can depend on this branch to work on the showcase and hopefully have things work in Pluto)

**Comment by rkurchin** [comment]
(lol, the old PR is failing tests now, but this rebased one passes, so I'm not gonna worry about it; we'll just merge and tag this and all should be good)

File: src/graph_building.jl (hunk 1 above)
Comment (thazhemadam) [reply src/graph_building.jl:142]: No, we can't, since `conn_matrix` here is a `PyObject` iirc.

**Comment by thazhemadam** [comment]
This looks good to be merged! Merging now.

**Pull request merged**